src/lib/components/Modal/_hooks/useModalFocus.js

```diff
@@ -0,0 +1,96 @@
+import { useEffect } from 'react';
+
+export const useModalFocus = (
+  autoFocus,
+  childrenWrapperRef,
+  closeButtonRef,
+  focusTrap,
+  primaryButtonRef,
+) => {
+  useEffect(
+    () => {
+      if (!autoFocus) {
+        return () => {};
+      }
+
+      // If `autoFocus` is set to `true`, following code finds all focusable elements and among them
+      // first not disabled form field element (input, textarea or select) or primary button
+      // and auto focuses it. This is necessary to have focus on one of those elements to be able
+      // to submit the form by pressing Enter key. If there are neither, it tries to focus any other
+      // focusable elements. In case there are none, childrenWrapperElement (Modal itself) is focused.
+
+      const childrenWrapperElement = childrenWrapperRef.current;
+
+      if (childrenWrapperElement == null) {
+        return () => {};
+      }
+
+      const childrenFocusableElements = Array.from(
+        childrenWrapperElement.querySelectorAll('button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'),
+      );
+
+      const firstFormFieldEl = childrenFocusableElements.find(
+        (element) => ['INPUT', 'TEXTAREA', 'SELECT'].includes(element.nodeName) && !element.disabled,
+      );
+
+      if (childrenFocusableElements.length === 0) {
+        childrenWrapperElement.tabIndex = 1;
+        childrenFocusableElements.push(childrenWrapperElement);
+      }
+
+      const firstFocusableElement = childrenFocusableElements[0];
+      const lastFocusableElement = childrenFocusableElements[childrenFocusableElements.length - 1];
+
+      if (firstFormFieldEl) {
+        firstFormFieldEl.focus();
+      } else if (primaryButtonRef?.current != null) {
+        primaryButtonRef.current.focus();
+      } else {
+        firstFocusableElement.focus();
+      }
+
+      const keyPressHandler = (e) => {
+        if (e.key === 'Escape' && closeButtonRef?.current != null) {
+          closeButtonRef.current.click();
+        }
+
+        if (
+          e.key === 'Enter'
+        && e.target.nodeName !== 'BUTTON'
+        && e.target.nodeName !== 'TEXTAREA'
+        && e.target.nodeName !== 'A'
+        && primaryButtonRef?.current != null
```

Comment:
The indentation looks weird 🤔 (hey ESlint!).

Reply:
Is this of elements final? Can we double check it with others? I am not sure if for example input type=button or things like this will work. @mbohal any idea or do you think that @hubacekj listed all options?

Reply:
I can not think of anything else. In the original implementation that @hubacekj is building upon we only checked for `button`, so this is an improvement already.

I think this is OK once the indentation is fixed.